src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+			command( javacCommand ).run();
+
+		} finally {
+			if ( FileExists( tempSrcFileName ) ){
+				fileDelete( tempSrcFileName );
+			}
+
+		}
+
+	}
+
+	function writeTempSourceFile( string tempSrcFileName , array sourcePath=getSourcePaths() , string extension=".java" ) {
+		var globber = wirebox.getInstance( 'globber' );
+
+		//shell.printString( " gSP-> #serialize(getSourcePaths())# " );
+        job.addLog( " gSP-> #serialize(sourcePath)# " );
+		var sourceList = globber
+				.setPattern( sourcePath )
+				.asQuery()
+				.matches()
+				.filter(( row ) => row.type=="file" && row.name.endsWith( extension ))
+				.reduce(( acc, row ) => {
+					return listappend( acc, row.directory & "/" & row.name, chr(10) );
+				}, "")
+
+        job.addLog( " sList-> #serialize(sourceList)# " );
+		if( !sourceList.len() ) {
+			/* throw(
+				message='No java source files found in [#sourceDirs.toList()#]', detail='Check fromSource() and try again',
+				type="commandException"
+			); */
+			throw(
+				message='No java source files found in [#sourceList.len()#]', detail='Check fromSource() and try again',
+				type="commandException"
+			);
+		}
+
+		fileWrite( tempSrcFileName, sourceList );
+	}
+
+	function buildJar() {
+		var currentLibsDir = fileSystemutil.resolvePath( getLibsDir(), getProjectRoot() );
+		var jarName = getJarNameString();
+        var currentProjectRoot = getProjectRoot();
+
+        var tempSrcFileName = tempDir & 'temp#createUUID()#.txt';
+
+		//var currentLibsDir = fileSystemutil.resolvePath( getLibsDir(), getProjectRoot() );
+		//shell.printString( ' cLD-> ' & currentLibsDir );
+
+        // if the jarname is empty
+        // we check if the current libs is a package
+        // if its a package
+        // get name from package
+        // if not get name from folder
+        // if its root
+        // give default name
+
+        job.start( ' for build jar check jarName ' );
+        if( jarName.len() == 0 ){
+            // jarName is empty
+            //shell.printString( ' jarName is empty ' );
+            job.addLog( ' jarName is empty ' );
+            job.addLog( ' currentLibsDir-> #currentLibsDir# ' );
+            job.addLog( ' currentProjectRoot-> #currentProjectRoot# ' );
+            if( packageService.isPackage( currentProjectRoot ) ) {
+                // its a package
+                //shell.printString( ' dir is a package ' );
+                job.addLog( ' dir is a package ' );
+                job.start( ' get jarName From Package ' );
+                jarName = getJarNameFromPackage( currentProjectRoot );
+                job.complete( getVerbose() );
+            } else {
+                // it is not a package its a normal folder
+                //shell.printString( ' dir is not a package ' );
+                job.addLog( ' dir is not a package ' );
+                var word = ListLast( getProjectRoot(), "\/" );
+                if( fileSystemUtil.isWindows() ) {
```

Comment:
The OS shouldn't matter here.  If `listLen( getProjectRoot(), '\/' )` is one or more, then do the `listLast()` bit, otherwise employ your default name.

Reply:
done